clippy_lints/src/utils/hir_utils.rs

```diff
@@ -691,23 +691,23 @@ impl<'a, 'tcx: 'a> SpanlessHash<'a, 'tcx> {
     pub fn hash_ty(&mut self, ty: &Ty) {
         std::mem::discriminant(&ty.node).hash(&mut self.s);
         match ty.node {
-            TyKind::Slice(ref ty) => {
```

Comment:
you have to use `ty::*` (lower case) and import `use rustc::ty::{self, TypeckTables};`